skipruntime-ts/helpers/src/utils.ts

```diff
@@ -68,3 +70,111 @@ export function reactiveResponseHeader(
 ): [string, string] {
   return ["Skip-Reactive-Response-Token", JSON.stringify(reactiveResponse)];
 }
+
+/*****************************************************************************/
+// Joins
+/*****************************************************************************/
+
+type JoinObject = { value: Json; side: "left" | "right" };
```

Comment:
I tried. But I didn't manage to make it work. Typescript is complaining with an error that I don't understand ...

Reply:
Let me try (I thought I tried to add JsonObject and it didn't work, but it might have been somewhere else).

Reply:
So yeah, typescript refuses it. I don't understand why ...

Reply:
I tried and pushed the result to your branch. Seems to work, or am I missing something?

Reply:
So I see you pushed a diff that makes JsonObject bubble up all the way to joinCollections. I don't know that this is a good idea.

The reason is that when I tried to use joinCollections on inputs, Typescript did not complain even when the input collections where not objects. That is a pretty subtle "gotcha" that I would like to avoid.